controllers/chat/moveToAnon.js

```diff
@@ -38,12 +44,30 @@ const moveToAnon = async (req, res) => {
   const filter = {id: oldChannelId};
   const sort = {last_message_at: -1};
   const oldChannelDetail = await client.queryChannels(filter, sort);
+  console.log('::oldChannelDetail', oldChannelDetail);
   //req.userId always anon since the token are using anon token
   const mySignedId = await UsersFunction.findSignedUserId(User, req.userId);
-  const prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
-    (user) => user.user_id !== mySignedId
-  )[0];
-  const detailPrevTargetUser = await UsersFunction.findUserById(User, prevTargetUser.user_id);
+
+  let prevTargetUser;
+  if (oldChannelDetail[0].data.better_channel_member) {
+    prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
+      (user) => user.user_id !== mySignedId
+    )[0];
+    targetUserId = prevTargetUser.user_id;
+  }
+
+  if (!prevTargetUser) {
+    prevTargetUser = {
+      user_id: targetUserId,
+      anon_user_info_color_code: color.code,
+      anon_user_info_color_name: color.color,
+      anon_user_info_emoji_code: emoji.emoji,
+      anon_user_info_emoji_name: emoji.name
+    };
+  }
+
+  const detailPrevTargetUser = await UsersFunction.findUserById(User, targetUserId);
+  console.log('::detailPrevTargetUser', detailPrevTargetUser);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are two console.log statements in this code block. While they may be useful for debugging during development, they should be removed or replaced with a proper logging mechanism before deploying to production. Also, the logic for assigning `prevTargetUser` and `targetUserId` seems complex and could potentially be simplified or broken down into smaller functions for better readability and maintainability.

```diff
- console.log('::oldChannelDetail', oldChannelDetail);
- console.log('::detailPrevTargetUser', detailPrevTargetUser);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
@brm-stnd kita mulai kurangi console log untuk di staging dan prod, kecuali ada case tertentuini tolong d remove.